### PR TITLE
[AA-1465] Add Applications API endpoint to Admin API

### PR DESCRIPTION
**Pull request by CSR2017**
@andonyns @pmcvtm 
Please help review and merge the changes. 
Thanks

### CODE REVIEW

**Review by pmcvtm — CHANGES_REQUESTED** [review]
This is mostly functional but we have a few important things to do before we merge:

- add e2e tests for new endpoints
- fix missing 404 responses
- fix up resource groupings and URLs
- explain or revert behavior changes in `Edit` command
- fix new build warnings

Please take a look at the other comments as well. We'll plan to address Swagger / documentation changes in another PR.

File: Application/EdFi.Ods.Admin.Api/Infrastructure/AutoMapper/AdminApiMappingProfile.cs [hunk 1]
```diff
@@ -25,6 +27,20 @@ public AdminApiMappingProfile()
                 .ForMember(dst => dst.ContactName, opt => opt.MapFrom(src => src.ContactName()))
                 .ForMember(dst => dst.ContactEmailAddress, opt => opt.MapFrom(src => src.ContactEmail()))
                 .ForMember(dst => dst.NamespacePrefixes, opt => opt.MapFrom(src => ToCommaSeparated(src.VendorNamespacePrefixes)));
+
+            CreateMap<Application, ApplicationModel>()
+                .ForMember(dst => dst.EducationOrganizationId, opt => opt.MapFrom(src => src.ApplicationEducationOrganizations.First().EducationOrganizationId))
+                .ForMember(dst => dst.ProfileName, opt => opt.MapFrom(src => src.ProfileName()));
```
[the remark this answers is not in the document]
Comment (pmcvtm) [reply Application/EdFi.Ods.Admin.Api/Infrastructure/AutoMapper/AdminApiMappingProfile.cs:33]: This answer here is _no_, or at least not easily! The `Application` class is in a different repo.

**Comment by pmcvtm** [comment]
I've addressed most of the feedback here, with the exception of Swagger changes we'll look at separately, and the missing e2e tests which I am still working on. @saa14 please give the code changes a look over. Thanks!

**Review by saa14 — APPROVED** [review]

**Review by pmcvtm — APPROVED** [review]
Should have re-approved with my other comment - LGTM, obviously.

We will merge this and open a second PR with new tests + repeatable patterns for the E2E Postman tests.

**Pull request merged**